functions.php

```diff
@@ -149,3 +147,186 @@ function get_lot($con, $id)
+
+        if ($image_mime !== 'image/jpeg' && $image_mime !== 'image/png') {
+            return "Файл должен быть в формате jpeg или png";
+        }
+    }
+}
+
+/**
+ * Проверяет поле начальной цены
+ * @param string Поле начальной цены
+ * @return string Сообщение об ошибке
+ */
+function validate_rate($field) {
+    if (strlen($_POST[$field]) == 0) {
+        return 'Поле не заполнено';
+    }
+
+    if ($_POST[$field] <= 0) {
+        return 'Цена должна быть больше ноля';
+    }
+}
+
+/**
+ * Проверяет поле шага ставки
+ * @param string Поле шага ставки
+ * @return string Сообщение об ошибке
+ */
+function validate_step($field) {
+    if (strlen($_POST[$field]) == 0) {
+        return 'Поле не заполнено';
+    }
+
+    if ($_POST[$field] <= 0 || is_float($_POST[$field])) {
+        return 'Шаг ставки должен быть целым числом больше ноля';
+    }
+}
+
+/**
+ * Проверяет поле даты окончания торгов
+ * @param string Поле даты
+ * @return string Сообщение об ошибке
+ */
+function validate_date($field) {
+    if (empty($_POST[$field])) {
+        return 'Поле не заполнено';
+    }
+
+    if (!is_date_valid($_POST[$field])) {
+        return 'Введите дату в формате ДД:ММ:ГГ';
+    }
+    if (strtotime($_POST[$field]) - time() + 3600 * 24 < 0) {
+        return 'Торги должны длиться минимум 24 часа';
+    }
+
+}
+
+/**
+ * Осуществляет валидацию формы
+ * @return array Массив с сообщениями об ошибках
+ */
+function validate_form () {
+    $errors = [];
+    $rules = [
+        'lot-name' => function() {
+            return validate_name('lot-name');
+        },
+
+        'message' => function() {
+            return validate_message('message');
+        },
+
+        'lot-image' =>  function() {
+            return validate_image('lot-image');
+        },
+
+        'lot-rate' =>  function() {
+            return validate_rate('lot-rate');
+        },
+
+        'lot-step' =>  function() {
+            return validate_step('lot-step');
+        },
+
+        'lot-date' =>  function() {
+            return validate_date('lot-date');
+        }
+    ];
+
+    foreach ($_POST as $key => $value) {
+        if (isset($rules[$key])) {
+            $rule = $rules[$key];
+            $errors[$key] = $rule();
+        }
+    }
+
+    foreach ($_FILES as $key => $value) {
+        if (isset($rules[$key])) {
+            $rule = $rules[$key];
+            $errors[$key] = $rule();
+        }
+    }
+
+
+    return array_filter($errors);
+
+}
+
+/**
+ * Перемещает загруженное изображение в директорию проекта
+ * @param string Поле файла
+ */
+function move_file($file) {
+    $file_name = $file['name'];
+    $file_path = __DIR__ . '/uploads/';
+    $file_url = '/uploads/' . $file_name;
+    move_uploaded_file($file['tmp_name'], $file_path . $file_name);
+    return $file_url;
+}
+
+function add_new_lot($con, $data) {
+    $lot_name = $data['lot-name'];
+    $description = $data['message'];
+    $lot_rate = settype($data['lot-rate'], 'integer');
```

Comment:
settype возвращает bool
если нужно поместить в новую переменную значение с приведением типа, то лучше это сделать более явно:
```suggestion
    $lot_rate = (int)$data['lot-rate'];
```